rs/index/src/vector/fixed_file.rs

```diff
@@ -117,7 +117,12 @@ impl<T: ToBytes + Clone> FixedFileVectorStorage<T> {
         for id in iterator {
             // Skip invalidated ids
             // TODO: Use skip list for better performance
-            if invalidated_ids.contains(&(id as u32)) {
+            let is_invalid_id = {
+                let guard = invalidated_ids.read().unwrap();
```

Comment:
no don't do this. grab the read lock once outside of the loop. each time you grab lock, it takes about 10ns. you don't want to do it in a hot loop